qiskit/circuit/quantumcircuit.py

```diff
@@ -2435,25 +2435,25 @@ def from_qasm_str(qasm_str: str) -> "QuantumCircuit":
 
     @property
     def global_phase(self) -> ParameterValueType:
-        """Return the global phase of the circuit in radians."""
+        """Return the global phase of the current circuit scope in radians."""
+        if self._control_flow_scopes:
+            return self._control_flow_scopes[-1].global_phase
```

Comment:
This is a necessary part of the interface being added in this PR, but is there really a use case for:

```python
with qc.while_loop((qc.clbits[1], False)):
    qc.h(0)
    print(qc.global_phase)
```

It seems pretty esoteric. But it's good we handle it

Reply:
Typically you wouldn't need to actually get the `global_phase` for your own use, but things like `__iadd__`, which is a fairly natural way of writing things, use both a get and a set by default.